tests/bdp_util.erl

```diff
@@ -0,0 +1,221 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+%% @doc Helper functions for bdp_service_manager.erl and others.
+
+-module(bdp_util).
+
+-export([build_cluster/1, build_cluster/2,
+         service_added/4, service_removed/2, service_started/4, service_stopped/4,
```

Comment:
These 4 function names seem misleading. The names indicate checking something that happened in the past or a current state, but they are actually attempting to do an action. I'd recommend the following changes to adhere to "least surprise":

* service_added -> add_service
* service_removed -> remove_service
* service_started -> start_service
* service_stopped -> stop_service